sagemaker_processing/feature_transformation_with_sagemaker_processing_Dask/container/program/bootstrap.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+import os
+import socket
+import json
+import subprocess
+import sys
+import time
+from shutil import copyfile
+from subprocess import Popen, PIPE
+DASK_PATH = "/opt/conda/bin/"
+
+
+def get_resource_config():
+    resource_config_path = "/opt/ml/config/resourceconfig.json"
+    with open(resource_config_path, "r") as f:
+        return json.load(f)
+
+def start_daemons(master_ip):
+    resource_config = get_resource_config()
+    current_host = resource_config["current_host"]
+    scheduler_host = resource_config["hosts"][0]
+
+    cmd_start_scheduler = DASK_PATH + "dask-scheduler"
+    cmd_start_worker = DASK_PATH + "dask-worker"
+
+    if current_host == scheduler_host:
+        Popen([cmd_start_scheduler])
+        Popen([cmd_start_worker, "tcp://" + str(master_ip) + ":8786"])
```

Comment:
@larroy Since these two lines will start a daemon in the background, there won't be a return code to check until the process terminates. I have added a check when connecting to the cluster which will error out if there is a problem with either the scheduler or any of the workers.

Do the processes need to be killed when the main script finishes?
You're absolutely right, it is not needed now as I added an alive check in the bootstap file.